WalletWasabi.Fluent/Models/Wallets/WalletRepository.cs

```diff
@@ -150,25 +162,52 @@ private async Task<IWalletSettingsModel> ImportWalletAsync(WalletCreationOptions
 
 	private async Task<IWalletSettingsModel> RecoverWalletAsync(WalletCreationOptions.RecoverWallet options)
 	{
-		var (walletName, password, mnemonic, minGapLimit) = options;
+		var (walletName, walletBackup, minGapLimit) = options;
 
 		ArgumentException.ThrowIfNullOrEmpty(walletName);
-		ArgumentNullException.ThrowIfNull(password);
-		ArgumentNullException.ThrowIfNull(mnemonic);
+
+		switch (walletBackup)
+		{
+			case RecoveryWordsBackup recoveryWordsBackup:
+				ArgumentNullException.ThrowIfNull(recoveryWordsBackup.Password);
+				ArgumentNullException.ThrowIfNull(recoveryWordsBackup.Mnemonic);
+				break;
+			case MultiShareBackup multiShareBackup:
+				ArgumentNullException.ThrowIfNull(multiShareBackup.Password);
+				ArgumentNullException.ThrowIfNull(multiShareBackup.Shares);
+				break;
+			default:
+				throw new ArgumentOutOfRangeException(nameof(walletBackup));
+		}
```

Comment:
These validations are here only because the properties are declared as nullable but they cannot really be null.